src/core/qgsdistancearea.cpp

```diff
@@ -347,6 +347,59 @@ double QgsDistanceArea::measureLine( const QgsPointXY &p1, const QgsPointXY &p2
   return result;
 }
 
+double QgsDistanceArea::measureLine3D( const QVector<QgsPoint> &points ) const
+{
+  if ( points.size() < 2 )
+    return 0;
+
+  double total = 0;
+  QgsPoint p1, p2;
+
+  try
+  {
+    p1 = points[0];
+
+    for ( QVector<QgsPoint>::const_iterator i = points.constBegin(); i != points.constEnd(); ++i )
+    {
+      p2 = *i;
+      total += measureLine3D( p1, p2 );
+
+      p1 = p2;
+    }
+
+    return total;
+  }
+  catch ( QgsCsException &cse )
+  {
+    Q_UNUSED( cse )
+    QgsMessageLog::logMessage( QObject::tr( "Caught a coordinate system exception while trying to transform a point. Unable to calculate line length." ) );
+    return 0.0;
+  }
+
+}
+
+double QgsDistanceArea::measureLine3D( const QgsPoint &p1, const QgsPoint &p2 ) const
+{
+  double result;
+
+  try
+  {
+    QgsPoint pp1 = p1, pp2 = p2;
+
+    QgsDebugMsgLevel( QStringLiteral( "Cartesian calculation on canvas coordinates" ), 4 );
+    result = p2.distance3D( p1 );
+  }
+  catch ( QgsCsException &cse )
```

Comment:
We don't do any CRS reprojection, so we don't need to catch exceptions here... (nor in the other measureLine3D())

Reply:
Function is removed now.

Reply:
function is removed.